viscy/representation/evaluation/lca.py

```diff
@@ -56,23 +64,53 @@ def fit_logistic_regression(
     tuple[LogisticRegression, tuple[tuple[NDArray, NDArray], tuple[NDArray, NDArray]]]
         Trained classifier and data split [[X_train, y_train], [X_test, y_test]].
     """
-    fov_selection = features["fov_name"].isin(train_fovs)
-    train_selection = fov_selection
-    test_selection = ~fov_selection
     annotations = annotations.cat.codes.values.copy()
+
+    # Handle background class removal before splitting for stratification
     if remove_background_class:
-        label_selection = annotations != 0
-        train_selection &= label_selection
-        test_selection &= label_selection
-        annotations -= 1
-    train_features = features.values[train_selection]
-    test_features = features.values[test_selection]
+        valid_indices = annotations != 0
+        features_filtered = features[valid_indices]
+        annotations_filtered = annotations[valid_indices] - 1
+    else:
+        features_filtered = features
+        annotations_filtered = annotations
+
+    # Determine train/test split
+    if train_fovs is not None:
+        fov_selection = features_filtered["fov_name"].isin(train_fovs)
+        train_selection = fov_selection
+        test_selection = ~fov_selection
+    else:
+        unique_fovs = features_filtered["fov_name"].unique()
+
+        fov_class_dist = []
+        for fov in unique_fovs:
+            fov_mask = features_filtered["fov_name"] == fov
+            fov_classes = annotations_filtered[fov_mask]
+            # Use majority class for stratification or class distribution
+            majority_class = pd.Series(fov_classes).mode()[0]
+            fov_class_dist.append(majority_class)
+
+        # Split FOVs, not individual samples
+        train_fovs, test_fovs = train_test_split(
+            unique_fovs,
+            test_size=1 - train_ratio,
+            stratify=fov_class_dist,
+            random_state=random_state,
+        )
+
+        # Create selection based on FOV assignment
+        train_selection = features_filtered["fov_name"].isin(train_fovs)
```

Comment:
This is duplicated with https://github.com/mehta-lab/VisCy/pull/273/files#diff-453d679b95a004f2f09ed3213b0a54d56c62c8c1d8da9dc94148205b9edb936bR80-R82